engine/class_modules/sc_hunter.cpp

```diff
@@ -5474,6 +5552,18 @@ struct wildfire_bomb_t: public hunter_spell_t
         wildfire_cluster -> execute();
     }
   }
+
+  void update_ready( timespan_t cd_duration ) override
+  {
+    if ( p() -> buffs.mad_bombardier_2 -> check() )
+    {
+      p() -> buffs.mad_bombardier_2 -> expire();
```

Comment:
Move buff expiration to the end of `wildfire_bomb_t::execute()`.

Reply:
Ughh... thinking about it we may actually need to have 2 separate buffs for 2pc and 4pc interaction with 2pc because of the way the spell is setup (both in game and in simc).

So, 2pc should be consumed inside `wildfire_bomb_t::execute()` to properly handle cd interaction, but for the purposes of 4pc interaction with 2pc `wildfire_bomb_t::impact()` the buff should actually linger until after the actual damage spell executes inside impact.
I think the easiest would be to do with 2 separate buffs tbh.

Ideally this crap should be tested in game once the PTR goes up.

Reply:
> Ughh... thinking about it we may actually need to have 2 separate buffs for 2pc and 4pc interaction with 2pc because of the way the spell is setup (both in game and in simc).

Yep, this is what I mean. There should be 1 buff for the cd reset and 1 buff for the increased damage. That's how I have implemented it currently with ``mad_bombardier_2`` and ``mad_bombardier_4``.

Reply:
Yeah, the list is still relevant for 4pc. But there are always diff way to skin a cat, I wrote what I was going to do for it.

Re different buffs - typically I'd just name the "main" one (the one you'd see in-game) as it's named in-game and append something to our "internal" ones. In this case you could name them `_2pc` and `_4pc` as they are used this way (`_2`/`_4` is a bit obscure).